ngx_sxg_utils.c

```diff
@@ -232,3 +235,118 @@ X509* load_x509_cert(const char* filepath) {
   fclose(certfile);
   return cert;
 }
+
+ngx_sxg_cert_chain_t ngx_sxg_empty_cert_chain() {
+  static ngx_sxg_cert_chain_t empty;
+  empty.serialized_cert_chain = sxg_empty_buffer();
+  empty.certificate = NULL;
+  empty.issuer = NULL;
+  empty.ocsp = NULL;
+  return empty;
+}
+
+void ngx_sxg_cert_chain_release(ngx_sxg_cert_chain_t* target) {
+  sxg_buffer_release(&target->serialized_cert_chain);
+  if (target->certificate != NULL) {
+    X509_free(target->certificate);
+  }
+  if (target->issuer != NULL) {
+    X509_free(target->issuer);
+  }
+}
+
+bool load_cert_chain(const char* cert_path, ngx_sxg_cert_chain_t* target) {
+  FILE* certfile = fopen(cert_path, "r");
+  if (!certfile) {
+    return false;
+  }
+  char passwd = 0;
+  X509* cert = PEM_read_X509(certfile, 0, 0, &passwd);
+  X509* issuer = PEM_read_X509(certfile, 0, 0, &passwd);
+  fclose(certfile);
+
+  target->certificate = cert;
+  target->issuer = issuer;
+  return (cert != NULL) && (issuer != NULL);
+}
+
+bool write_cert_chain(ngx_sxg_cert_chain_t* cert,
+                      sxg_buffer_t* dst) {
+  sxg_buffer_t sct_list = sxg_empty_buffer();
```

Comment:
The lifetime of `sct_list` confuses me.  Do you need to call `sxg_buffer_release`?

Reply:
Sorry for confusing, libsxg has functionality to embed sct_list in the chain, but this module does not use it. Passing *empty* list is the only way to bypassing the functionality.
Then, `sct_list` must be always empty, empty buffer does not require `sxg_buffer_release`.
I'll rename it to `empty_sct_list`.